clustering_key_filter.hh

```diff
@@ -56,6 +56,10 @@ public:
     size_t size() const { return _ref.get().size(); }
     const clustering_row_ranges& ranges() const { return _ref; }
 
+    // Returns all clustering ranges determined by `slice` inside partition determined by `key`.
+    // If the slice contains the `reversed` option, we assume that it is given in 'half-reversed' format
+    // (i.e. the ranges within are given in reverse order, but the ranges themselves are not reversed).
+    // The ranges will be returned in forward (increasing) order even if the slice is reversed.
```

Comment:
This is great, but still ambiguous, because "forward" and "reversed" depend on the context (query order or table order). 

Please us unambiguous terms, I propose table order and query order.

Reply:
> "forward" and "reversed" depend on the context (query order or table order)

Not really. The function doesn't really depend on any queries or tables, it's just a simple data transformation on a slice, it can be understood outside any context.

It's all unambiguous when we consider `schema`, which is also passed in as a parameter to this function. A slice may be 'forward', 'half-reversed', or 'native-reversed' (not used here) w.r.t a specific `schema`, and that can only be interpreted in one way.

I can try improving the comment by saying "w.r.t `schema`" explicitly.